In [11]:
import numpy as np
from math import *

# Задача 1

In [13]:
M = 80
n = 256
s = sqrt(16/n)
T2 = M + 1.96*s
T2

80.49

In [14]:
T1 = M - 1.96*s
T1

79.51

95%  значений выборки находятся в интервале от 79.51 до 80.49

# Задача 2

In [15]:
a=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [16]:
Mx=a.mean()
Mx

6.590000000000001

In [17]:
s0=np.std(a, axis=None, dtype=None, out=None, ddof=1)
s0

0.4508017549014448

In [18]:
# критерий Стьюдента для 95% доверительного итервала (а/2 = 0,025)
t = 2.8214
T2 = Mx+t*s0/sqrt(a.size) 
T2

6.992207588315067

In [19]:
T1 = Mx-t*s0/sqrt(a.size) 
T1

6.187792411684934

доверительный интервал 95% для данной выборки для данной выборки имеет границы от 6.19 до 6.99

# Задача 3

In [21]:
# H0 - мю (1) = мю(0) 
# H1 - мю (1) > мю(0)
X = 17.5
M = 17
z_95% = 1.645
D = 4
n = 100

In [22]:
zn = (X-M)/sqrt(4/100)
zn

2.5

вывод: Z- наблюдаемое = 2,5 что значительно превышает критическое значение Z(95%) = 1.645
вывод: гипотеза Н0 не верна, мю(1) > мю(0)

# Задача 4

Н0 - мю(0)=200;  Н1 - мю != 200

In [26]:
a=np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [27]:
M0 = 200
Mx=a.mean()
Mx

198.5

In [28]:
s0=np.std(a, axis=None, dtype=None, out=None, ddof=1)
s0

4.453463071962462

In [29]:
# критерий Стьюдента для 99% доверительного итервала (а/2 = 0,005)
t = 3.25

In [30]:
T1 = Mx+t*s0/sqrt(a.size) 
T1

203.6962365302688

In [31]:
T2 = Mx-t*s0/sqrt(a.size) 
T2

193.3037634697312

среднее М(Х) для генеральной совокупности с вероятностью 99% располагается в диапазоне от 194 до 203 граммов, таким образом утверждение продавца может соответствовать действительности

In [32]:
# критерий Стьюдента для 95% доверительного итервала (а/2 = 0,025)
t = 2.8214
T1 = Mx+t*s0/sqrt(a.size) 
T1

202.4734021049138

In [33]:
T2 = Mx-t*s0/sqrt(a.size) 
T2

194.5265978950862

уменьшив верояность до 95%, получаем более узкий доверительный интервал с границами от 194 до 202 граммов. но и в этом случае утверждение продавца укладывается в возмножые значения.
Вывод: продавец говорит правду.

In [34]:
s0*2.58

11.489934725663153

In [36]:
tn=(Mx-M0)/(s0/sqrt(a.size))
tn

-1.0651074037450896

H0 верна для значений t от -3.25 до + 3.25
наблюдаемое значение tn = -1.065 укладывается в этот интервал
вывод: гипотеза Н0 верна, мю = 200